VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -187,7 +186,17 @@ public static CharSequence trim(CharSequence s) {
     }
 
     public static CharSequence removeDuplicateWhitespace(CharSequence text) {
-        return PATTERN_DUPLICATE_WHITESPACE.matcher(text).replaceAll(" ");
+        SpannableStringBuilder builder = new SpannableStringBuilder(text);
+        for (int size = text.length(), i = 0; i < size; i++) {
+            if (Character.isWhitespace(builder.charAt(i))) {
+                int nextIndex = i + 1;
+                if (nextIndex < size && Character.isWhitespace(builder.charAt(nextIndex))) {
+                    builder.replace(i, nextIndex, " ");
```

Comment:
Did you test this? I think the ending index is exclusive, and as such should be `nextIndex + 1`.